### ISSUE TITLE
Система торговли

### ISSUE THREAD

**Issue by Chief01** [Zealot111/SW_MissionExample#8]
Система торговли должна действовать при взаиодействии с торговцем либо с НПС, так же возможен гибридный вариант. НПС - ящик с которым производится торговля, торговец отдельный РП персонаж который выдает задания и общается со сталкерами.

Реализация :
При нажатии сочетания кнопок (или из выпадающего списка) вызывать интерфейс торговли, на котором будет отображено две кнопки покупка либо продажа а так же отмена. При покупке открывается список продаваемого снаряжения и вооружения. выбрав необходимое игрок нажимает купить и если  у него достаточно средств он покупает. После соверщения сделки снаряжение спавнится в ящике. Для совершения продажи игрок складывает предметы в ящик и нажимает продать.

**Comment by ghost** [comment]
Я думаю, что ящик тут лишнее - достаточно проверить место в рюкзаке и класть туда. Также продавать из рюкзака\инвентаря.
Каждую шмотку покупать индивидуально - выбрал - есть деньги? есть место? купил.

**Comment by Zealot111** [comment]
> Я думаю, что ящик тут лишнее - достаточно проверить место в рюкзаке и класть туда. Также продавать из рюкзака\инвентаря.

Это неудобно. Там может быть мало место, оттуда вещь могут украсть.

**Comment by ghost** [comment]
Да, из рюкзака крадут, даже я на тф рп кооп воровал на станции Янов у кого-то бинты :)
1) Решение тогда непонятно - если человек накупил, а места у него не хватает, что ему делать?
2) как сделать у каждого свой ящик?

Может попробовать блокировать на время рюкзак покупателя?
Проверки
canAddItemToBackpack
canAddItemToUniform
canAddItemToVest

**Comment by Zealot111** [comment]
> 2) как сделать у каждого свой ящик?

Создавать его локальным. Тогда просто у каждого пользователя будет свой ящик на том же саммо месте. И вещи в нем могут хранить в течении игры.
